File: python/binary-search/find-minimum-in-rotated-sorted-array.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        left, right = 0, len(nums) - 1
        answer = float('inf')
        while left <= right:
            middle = (left + right) // 2
            answer = min(answer, nums[middle])
            if nums[middle] > nums[right]:
                left = middle + 1
            else:
                right = middle - 1
        return answer

File: c++/binary-search/find-minimum-in-rotated-sorted-array.cc
class Solution {
public:
    int findMin(vector<int>& nums) {
        int left = 0;
        int right = nums.size() - 1;

        int answer = std::numeric_limits<int>::max();
        while (left <= right) {
            int middle = (left + right) / 2;
            answer = std::min(answer, nums[middle]);

            if (nums[left] <= nums[middle] && nums[right] < nums[left]) {
                left = middle + 1;
            } else {
                right = middle - 1;
            }
        }
        return answer;
    }
};
